File: app/src/main/java/com/enipro/data/remote/model/SessionTiming.java
package com.enipro.data.remote.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * A session timing is basically at what time during a day a session will hold.
 * e.g 10am to 12pm on Mondays
 * The time in from and to are in 24 hour mode.
 */
@org.parceler.Parcel
public class SessionTiming {

    @SerializedName("from_time")
    @Expose
    public String time_from;

    @SerializedName("to_time")
    @Expose
    public String time_to;

    @SerializedName("days")
    @Expose
    public List<String> days;

    public SessionTiming() {

    }


    public SessionTiming(List<String> days, String from, String to) {
        this.days = days;
        this.time_from = from;
        this.time_to = to;
    }

    public String getFrom() {
        return time_from;
    }

    public List<String> getDays() {
        return days;
    }

    public String getTo() {
        return time_to;
    }

    public void setDays(List<String> days) {
        this.days = days;
    }

    public void setFrom(String from) {
        this.time_from = from;
    }

    public void setTo(String to) {
        this.time_to = to;
    }
    
}

File: app/src/main/java/com/enipro/data/remote/model/SessionTiming.kt
package com.enipro.data.remote.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class SessionTiming(
        @field:SerializedName("days")
        @field:Expose var days: List<String>,
        @field:SerializedName("from_time")
        @field:Expose var from: String,
        @field:SerializedName("to_time")
        @field:Expose var to: String) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.createStringArrayList(),
            parcel.readString(),
            parcel.readString())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeStringList(days)
        parcel.writeString(from)
        parcel.writeString(to)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<SessionTiming> {
        override fun createFromParcel(parcel: Parcel): SessionTiming {
            return SessionTiming(parcel)
        }

        override fun newArray(size: Int): Array<SessionTiming?> {
            return arrayOfNulls(size)
        }
    }
}
